frontend/src/components/authentication/CreateAccountPage/CreateAccountPage.tsx

```diff
@@ -118,112 +128,147 @@ const CreateAccountPage = (): JSX.Element => {
+        Return: boolean (true if valid, false if not)
+        */
+        //setErrorMessages({...errorMessagesInitial});
+        if (firstName === "") {
+            setNameError("Please enter your first name");
+            return false;
+        } 
+        setNameError("");
+        return true;
+    }
+
+    const validateLastName = (lastName: string): boolean => {
+        /*
+        Desc: Validates firstName and lastName
+        Return: boolean (true if valid, false if not)
+        */
+        //setErrorMessages({...errorMessagesInitial});
+        if (lastName === "") {
+            setNameError("Please enter your last name");
+            return false;
+        } 
+        setNameError("");
         return true;
     }
 
-    const validatePhoneNumber = (): boolean => {
+    const validatePhoneNumber = (phoneNumber: string): boolean => {
         /*
         Desc: Validates phone number
         Return: boolean (true if valid, false if not)
         */
         if (phoneNumber === "") {
-            alert("Please add a phone number");
+            setPhoneNumberError("Please enter a phone number");
             return false;
         }
+        if(!processPhoneNumber(phoneNumber)) {
+            return false;
+        }
+        setPhoneNumberError("");
         return true;
     }
 
-    const validateEmail = (): boolean => {
+    const validateEmail = (email: string): boolean => {
         /*
         Desc: Validates email
         Return: boolean (true if valid, false if not)
         */
         if (email === "") {
-            alert("Please add email");
+            setEmailError("Please enter your email");
             return false;
         }
         else if (!email.includes("@")) {
-            alert("Please enter a valid email address");
+            setEmailError("Please enter a valid email address");
             return false;
         }
         //else if (check if email already exists)
         //alert("Account with this email already exists") 
+        setEmailError("");
         return true;
     }
 
-    const validatePassword = (): boolean => {
+    const validatePassword = (password: { value: string; }): boolean => {
         /*
         Desc: Validates password
         Return: boolean (true if valid, false if not)
         */
         const MIN_PASSWORD_LENGTH = 6;
         if (password.value === "") {
-            alert("Please add password");
+            setPasswordError("Please enter a password");
             return false;
         }
         else if (password.value.length < MIN_PASSWORD_LENGTH) {
-            alert(`Please choose a password at least ${MIN_PASSWORD_LENGTH} characters long`);
+            setPasswordError(`Please choose a password at least ${MIN_PASSWORD_LENGTH} characters long`);
             return false;
         }
+        setPasswordError("");
         return true;
     }
 
-    function processPhoneNumber(): boolean {
+    function processPhoneNumber(phoneNumber: string): boolean {
```

Comment:
Please check how thorough this function is. I would also add in another function from the validator package mentioned above:
using the `isMobilePhone(str [, locale [, options]])` function

Reply:
With isMobilePhone I set it to only accept US formatted numbers (e.g. (XXX)-XXX-XXXX or 1-(XXX)-XXX-XXXX).

Reply:
Yes, that should be fine